src/metabase/api/search.clj

```diff
@@ -477,26 +431,38 @@
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
 (mu/defn ^:private search-context :- SearchContext
-  [search-string   :- [:maybe ms/NonBlankString]
-   archived-string :- [:maybe ms/BooleanString]
-   table-db-id     :- [:maybe ms/PositiveInt]
-   models          :- [:maybe [:or SearchableModel [:sequential SearchableModel]]]
-   limit           :- [:maybe ms/PositiveInt]
-   offset          :- [:maybe ms/IntGreaterThanOrEqualToZero]]
-  (cond-> {:search-string      search-string
-           :archived?          (Boolean/parseBoolean archived-string)
-           :current-user-perms @api/*current-user-permissions-set*}
-    (some? table-db-id) (assoc :table-db-id table-db-id)
-    (some? models)      (assoc :models
-                               (apply hash-set (if (vector? models) models [models])))
-    (some? limit)       (assoc :limit-int limit)
-    (some? offset)      (assoc :offset-int offset)))
+  [{:keys [archived-string
+           created-by
+           limit
+           models
+           offset
+           search-string
+           table-db-id]} :- [:map {:closed true}
+                             [:search-string                    [:maybe ms/NonBlankString]]
+                             [:archived-string {:optional true} [:maybe ms/BooleanString]]
+                             [:created-by      {:optional true} [:maybe ms/PositiveInt]]
+                             [:limit           {:optional true} [:maybe ms/Int]]
+                             [:models          {:optional true} [:maybe [:or SearchableModel [:sequential SearchableModel]]]]
+                             [:offset          {:optional true} [:maybe ms/Int]]
+                             [:table-db-id     {:optional true} [:maybe ms/PositiveInt]]]]
+  (let [models (if (string? models) [models] models)
+        ctx    (cond-> {:search-string      search-string
+                        :current-user-perms @api/*current-user-permissions-set*
+                        :archived?          (Boolean/parseBoolean archived-string)
+                        :models             (apply hash-set (if (seq models) models search-config/all-models))}
```

Comment:
`:models` is being replaced on line 457
```suggestion
```

Reply:
Also, you can make `search-config/all-models` private now since it is otherwise not used outside the search-config namespace
https://github.com/metabase/metabase/blob/fc2f20fe5df0e4cdee99281c075f6e2779f6e0c8/src/metabase/search/config.clj#L69

Reply:
the models on line 452 is necessary because it's used as the starting point for `search-context->applicalble-models`

Reply:
Got it. Based on the spec we still need to handle the case where `:models` in the args is a single `SearchableModel` rather than a vector. Though perhaps it's cleaner to handle that in the `defendpoint` and leave this function to just take a `[:sequential SearchableModel]`

Reply:
yep. made the change in https://github.com/metabase/metabase/pull/32469/commits/91f41fb37faa6aba5c6115d3b948e8dfee6693b3